src/Http/WebUtilities/src/MultipartPipeReaderStream.cs

```diff
@@ -0,0 +1,348 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    class MultipartPipeReaderStream : Stream
+    {
+        private readonly PipeReader _pipeReader;
+        private long _observedLength = 0;
+        private bool _finished = false;
+        private bool _metadataSkipped = false;
+        private readonly MultipartBoundary _boundary;
+        private int _partialMatchIndex = 0;
+        private readonly bool _canSeek;
+
+        private static ReadOnlySpan<byte> CrlfDelimiter => new byte[] { (byte)'\r', (byte)'\n' };
+        private static ReadOnlySpan<byte> EndOfFileDelimiter => new byte[] { (byte)'-', (byte)'-' };
+
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => _canSeek;
```

Comment:
This doesn't seem right, Seek still throws. Why was this necessary?

Reply:
Check my response about it from your other comment. It's wrong, but it kept me from changing another file before ensuring my logic is correct.